aggsender/query/certificate_query.go

```diff
@@ -65,17 +66,13 @@ func (c *certificateQuerier) GetLastSettledCertificateToBlock(
 	)
 
 	// 1. Get the latest settled bridge exit block number
-	if cert.NewLocalExitRoot != types.EmptyLER {
-		// if NewLER is not the first empty LER, it means that the certificate
-		// or certificate before it had bridge exits, so we can use it to
-		// to determine the last bridge exit block
-		newLER, err := c.l2BridgeSyncer.GetExitRootByHash(ctx, cert.NewLocalExitRoot)
-		if err != nil {
-			return 0, fmt.Errorf("failed to get exit root by hash using NewLocalExitRoot %s: %w",
-				cert.NewLocalExitRoot.String(), err)
-		}
-
-		lastBridgeExitBlock = newLER.BlockNum
+	// if NewLER is not the first empty LER, it means that the certificate
+	// or certificate before it had bridge exits, so we can use it to
+	// to determine the last bridge exit block
+	lastBridgeExitBlock, err = c.getBlockNumFromLER(ctx, cert.NewLocalExitRoot)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get exit root by hash using NewLocalExitRoot %s: %w",
```

Comment:
This error message needs to be adapted, something like


```suggestion
		return 0, fmt.Errorf("failed to resolve the bridge exit block number for NewLocalExitRoot %s: %w",
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/863/commits/c26ec0fc78d2e5b08ad3c9667e4a78f3af6b5668